plugins/inventory/azure_rm.py

```diff
@@ -444,6 +468,7 @@ def _on_arcvm_page_response(self, response):
             url = '{0}/providers/Microsoft.AzureStackHCI/virtualMachineInstances'.format(arcvm['id'])
             # Stack HCI instances look close enough to regular VMs that we can share the handler impl...
             self._enqueue_get(url=url, api_version=self._stackhci_api_version, handler=self._on_vm_page_response, handler_args=dict(arcvm=arcvm))
+            self._hosts.append(ArcHost(arcvm, self, legacy_name=self._legacy_hostnames))
```

Comment:
this causes me an issue in the scenario where :
- this is an azure stack hci vm
- i'm using a custom hostvar_expression in inventory.azure_rm.yml
```
hostvar_expressions:
  ansible_host: (private_ipv4_addresses | first) if os_profile.system == 'linux' else computer_name | default(name, true)
```

it breaks the plugin with:
```
[WARNING]:  * Failed to parse /mnt/c/Git/CoreIT/ansible/ansible-windows/inventories/HCI/sam/inventory.azure_rm.yml with
auto plugin: Could not set ansible_host for host S000004BK4: 'private_ipv4_addresses' is undefined
```

because indeed, no `private_ipv4_addresses` property on the ArcHost object, but there is one on the AzureHost object (stack hci vm).
if i comment the line, the inventory works fine for me (vm present and with proper value for ansible_host, according to my host var expression)

the AzureHost should superseed the ArcHost object because it as more properties
i'm not sure how the merge occurs though

Reply:
This is done